scriptworker/cot/verify.py

```diff
@@ -960,111 +916,9 @@ def verify_link_in_task_graph(chain, decision_link, task_link):
         verify_task_in_task_graph(task_link, graph_defn)
         log.info("Found {} in the graph; it's a match".format(task_link.task_id))
         return
-    # Fall back to fuzzy matching to support retriggers: the taskId and
-    # datestrings will change but the task definition shouldn't.
-    for task_id, graph_defn in decision_link.task_graph.items():
-        log.debug("Fuzzy matching against {} ...".format(task_id))
-        try:
-            verify_task_in_task_graph(task_link, graph_defn, level=logging.DEBUG)
-            log.info("Found a {} fuzzy match with {} ...".format(task_link.task_id, task_id))
-            return
-        except CoTError:
-            pass
-    else:
-        raise_on_errors(["Can't find task {} {} in {} {} task-graph.json!".format(
-            task_link.name, task_link.task_id, decision_link.name, decision_link.task_id
-        )])
-
-
-# verify_decision_command {{{1
-def verify_decision_command(decision_link, mach_commands=DECISION_MACH_COMMANDS):
-    r"""Verify the decision command for a decision task.
-
-    This is deprecated, but still very much in use.
-    Let's stop needing it.
-
-    Some example commands::
-
-        "/builds/worker/bin/run-task",
-        "--vcs-checkout=/builds/worker/checkouts/gecko",
-        "--",
-        "bash",
-        "-cx",
-        "cd /builds/worker/checkouts/gecko &&
-        ln -s /builds/worker/artifacts artifacts &&
-        ./mach --log-no-times taskgraph decision --pushlog-id='83445' --pushdate='1478146854'
-        --project='mozilla-inbound' --message=' ' --owner='cpeterson@mozilla.com' --level='3'
-        --base-repository='https://hg.mozilla.org/mozilla-central'
-        --head-repository='https://hg.mozilla.org/integration/mozilla-inbound/'
-        --head-ref='e7023fe48f7c48e33ef3b91747647f0873e306d6'
-        --head-rev='e7023fe48f7c48e33ef3b91747647f0873e306d6'
-        --revision-hash='e3e8f6327079496707658adc381c142c6575b280'\n"
-
-        "/builds/worker/bin/run-task",
-        "--vcs-checkout=/builds/worker/checkouts/gecko",
-        "--",
-        "bash",
-        "-cx",
-        "cd /builds/worker/checkouts/gecko &&
-        ln -s /builds/worker/artifacts artifacts &&
-        ./mach --log-no-times taskgraph decision --pushlog-id='0' --pushdate='0' --project='date'
-        --message='try: -b o -p foo -u none -t none' --owner='amiyaguchi@mozilla.com' --level='2'
-        --base-repository=$GECKO_BASE_REPOSITORY --head-repository=$GECKO_HEAD_REPOSITORY
-        --head-ref=$GECKO_HEAD_REF --head-rev=$GECKO_HEAD_REV --revision-hash=$GECKO_HEAD_REV
-        --triggered-by='nightly' --target-tasks-method='nightly_linux'\n"
-
-    This is very firefox-centric and potentially fragile, but decision tasks are
-    important enough to need to monitor.  The ideal fix would maybe be to simplify
```

Comment:
What has changed that we don't care about this anymore?

Reply:
This was one of the fragile and weakest parts of CoT v1, looking at the decision task and trying to spot check it to make sure that it looked vaguely like a decision task. It's complex, fragile, and doesn't actually protect us from all decision task changes. CoT v2 added json-e rebuilding of the decision task, so we verify the entirety of the decision task comes from `.taskcluster.yml`, not just the command. [bug 1395413](https://bugzilla.mozilla.org/show_bug.cgi?id=1395413) also describes some reasons why this command check is not ideal.

This was another part of CoT v1 that we were very interested in sunsetting when everything supported CoT v2.